src/hooks/useTokenBalances.ts

```diff
@@ -57,7 +59,9 @@ async function fetchBalancesForToken(
     totalExchangeBalance: calculateTotalBalance(exchangeBalance, currentBatchId, pendingDeposit),
     pendingDeposit,
     pendingWithdraw,
-    claimable: pendingWithdraw.amount.isZero() ? false : pendingWithdraw.batchId < currentBatchId,
+    claimable: pendingWithdraw.amount.isZero()
+      ? false
+      : lastCreditBatchId !== currentBatchId && pendingWithdraw.batchId < currentBatchId,
```

Comment:
I think it's good to expose also `creditedOnLastBatch` and `withdrawHasMatured` so we can handle it better in the UI